astropy/coordinates/tests/test_regression.py

```diff
@@ -600,3 +602,28 @@ def test_regression_8138():
     newframe = GCRS()
     sc2 = sc.transform_to(newframe)
     assert newframe.is_equivalent_frame(sc2.frame)
+
+
+def test_regression_8276():
+    from astropy.coordinates import baseframe
+
+    with pytest.raises(TypeError) as excinfo:
+        class MyFrame(BaseCoordinateFrame):
+            a = QuantityAttribute(unit=u.m)
```

Comment:
This will now raise an error, so the rest of the test will not get executed! You may still be able to do it by using a `TimeAttribute` with an invalid time string as a default or so.

Reply:
That's intended: the idea is that *if* the more substantial change in #8276 goes in, this test should be run *without* the `pytest.raises` part.  I should probably leave a comment in the code saying that, though, now that you mention it!